src/kudu/consensus/consensus_queue.cc

```diff
@@ -999,8 +999,62 @@ void PeerMessageQueue::AdvanceMajorityReplicatedWatermarkFlexiRaft(
 }
 
 std::map<std::string, int> PeerMessageQueue::GetDataCommitQuorum() {
+  // Map to store the number of voters in each region from the active config.
+  std::map<std::string, int> voter_distribution;
+
+  // Return value denoting the number of votes required in each region.
   std::map<std::string, int> data_commit_quorum;
-  // TODO(ritwikyadav) : Populate quorum from cmeta_
+
+  CHECK(queue_state_.active_config->has_commit_rule());
+
+  // Set to help process each peer only once.
+  std::set<std::string> unique_uuids;
+
+  // Step 1: Compute total number of voters in each region.
+  for (const RaftPeerPB& peer : queue_state_.active_config->peers()) {
+    // Only consider peers that are voters.
+    if (!peer.has_member_type() ||
+        peer.member_type() != RaftPeerPB::VOTER) {
+      continue;
+    }
+    CHECK(peer.has_permanent_uuid() && peer.has_attrs() &&
+        peer.attrs().has_region());
+    const std::string& region = peer.attrs().region();
+    if (unique_uuids.find(peer.permanent_uuid()) == unique_uuids.end()) {
+      int& count = LookupOrInsert(&voter_distribution, region, 0);
+      count++;
+      unique_uuids.insert(peer.permanent_uuid());
+    }
+  }
+
+  // Step 2: Compute data commit quorum from the config.
+  for (const RegionRuleSpecPB& rule_spec :
+      queue_state_.active_config->commit_rule().rule_conjunctions()) {
+    std::string region = rule_spec.region_identifier();
+    const std::string& commit_req = rule_spec.commit_requirement();
+
+    // Resolve master region.
+    if (region == "master") {
+      CHECK(local_peer_pb_.has_attrs() && local_peer_pb_.attrs().has_region());
+      region = local_peer_pb_.attrs().region();
+    }
+
+    // TODO(ritwikyadav): It is possible that all voters in a region
+    // have left the replicaset. Handle that case gracefully in a subsequent diff.
+    int total_voters = FindOrDie(voter_distribution, region);
+    int region_majority = total_voters / 2 + 1;
+
+    int commit_req_num = -1;
+
+    // Resolve majority specification.
```

Comment:
this should become  afunction, because comparison with string majority needs to be hidden from callers.